File: Data Structures & Algorithms/two-integer-sum/submission-2.js
class Solution {
    /**
     * @param {number[]} nums
     * @param {number} target
     * @return {number[]}
     */
    twoSum(nums, target) {
        let left = 0;
        let right = nums.length - 1;
        console.log(nums)
        while ( left < right ) {
            let sum = nums[left] + nums[right]
            console.log({sum, left: nums[left], right: nums[right]})
            if(sum > target) right--;
            else if (sum < target) left++;
            else return [left, right]
        }
    }
}

File: Data Structures & Algorithms/three-integer-sum/submission-2.py
class Solution:
    def threeSum(self, nums: List[int]) -> List[List[int]]:
        result = set()
        nums = sorted(nums)
        for i in range(0,len(nums) - 2):
            j = i + 1
            k = len(nums) - 1
            while j < k:
                _sum = nums[i] + nums[j] + nums[k]
                if _sum == 0:
                    result.add((nums[i],nums[j],nums[k]))
                    j += 1
                    k -= 1
                elif _sum < 0:
                    j += 1
                else:
                    k -= 1
        return [list(x) for x in result]